lib/tree-view.coffee

```diff
@@ -846,9 +846,49 @@ class TreeView
         repo.getPathStatus(initialPath)
         repo.getPathStatus(newPath)
 
+    catch error
+      if error.code is 'EEXIST'
+        return @moveConflictingEntry(initialPath, newPath, newDirectoryPath)
+      else
+        @emitter.emit 'move-entry-failed', {initialPath, newPath}
+        atom.notifications.addWarning("Failed to move entry #{initialPath} to #{newDirectoryPath}", detail: error.message)
+
+    return true
+
+  moveConflictingEntry: (initialPath, newPath, newDirectoryPath) =>
+    try
+      if fs.isFileSync(initialPath)
```

Comment:
You may want to check if it's not a directory instead, to avoid handling symlinks as directories:

```suggestion
      unless fs.isDirectorySync(initialPath)
```